ext/mini_racer_extension/mini_racer_extension.cc

```diff
@@ -225,7 +225,13 @@ static Handle<Value> convert_ruby_to_v8(Isolate* isolate, VALUE value) {
 
     switch (TYPE(value)) {
     case T_FIXNUM:
-	return scope.Escape(Integer::New(isolate, NUM2INT(value)));
+        length = NUM2INT(rb_funcall(value, rb_intern("size"), 0));
+        if (length > 4)
+        {
+            return scope.Escape(Number::New(isolate, NUM2DBL(value)));
```

Comment:
I see, so we lose some precision here... ok, my call is to use NUM2LONG, then we avoid the funccall, simply check if its larger than int max value if larger then convert to double etc...